README.md

```diff
@@ -1391,3 +1392,21 @@ documentation](https://docs.rubocop.org/rubocop/configuration.html#inheriting-co
     do_something
   end
   ~~~
+  
+## Memoization
+
+* Prefer `return @x if defined?(@x)` over a simple `||=`
```

Comment:
```suggestion
* Prefer `return @x if defined?(@x)` over `||=` if the statement being memoized can evaluate to `nil` or `false`.
```

Reply:
Frankly, everything around memoization requires nuance, because memoization is caching and cache invalidation is one of the Two Hard Problems[^1]. I think your addition is a good detail to include in our style guide to reduce the chance someone mindlessly repeats the rose memoization pattern. But it would also be suboptimal for someone to mindlessly repeat `return @x if defined?(@x)`.

[^1]: along with naming things and off-by-one errors, famously

I don't support _forbidding_ rose memoization, so I'll advocate for the nuance in our discussion of it.

---

On a similar note, a section on Memoization might also include a note about memoizing a method that takes arguments.

```
# Bad. A stale value can be returned when a different argument is passed for subsequent calls
def expensive_result(input)
  @expensive_result ||= ExpensiveResult.calculate(input)
end

# Better. Recalculates when a different argument is passed.
# Consider using an LRU or not memoizing at all if the argument cardinality
# is high or the object is long-lived.
def expensive_result(input)
  @expensive_results ||= {}
  @expensive_results[input] ||= ExpensiveResult.calculate(input)
end
```